SOF-6896: fix pseudopotential source definition

pranabdas opened the pull request:
Make schema definition in consitent with current web-app schema

pranabdas committed:
SOF-6896: fix pseudopotential source definition

Make schema definition in consitent with current web-app schema

pranabdas committed:
SOF-6896: make categories non-required property in unit_method

timurbazhirov commented on example/methods_directory/physical/psp.json:
```diff
@@ -1,49 +1,52 @@
     },
     "data": {
-        "pseudo": [
+        "element": "Ru",
```
"data" has to contain an array. We can replace "pseudo" with "metaProperties".

pranabdas replied:
Ok, need to refactor in `web-app/src/application/imports/properties/server/startup.js`

timurbazhirov commented on example/methods_directory/physical/psp.json:
```diff
@@ -1,49 +1,52 @@
+        "type": "nc",
+        "version": "0.4"
+    },
+    "source": { "info": {}, "type": "exabyte" }
```
This source doesn't make sense, there's already one on line 47

pranabdas replied:
Seems this is not used (`web-app/src/application/imports/properties/server/startup.js`), shall we get rid of it?

timurbazhirov commented on schema/methods_directory/physical/psp.json:
```diff
@@ -23,7 +27,20 @@
+            "properties": {
+                "info": {
+                    "type": "object",
+                    "description": "currently an empty object"
```
See above

pranabdas committed:
SOF-6896: remove unnecessary source key and reformat pseudo data under metaProperties

pranabdas committed:
SOF-6896: flatten metaProperties

+ store psedo contents directly in data array
+ keep name and method path keys
+ keep category required

timurbazhirov committed:
Merge branch 'dev' into fix/SOF-6896
